sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -288,9 +288,10 @@ case class TruncateTableCommand(
   override def run(sparkSession: SparkSession): Seq[Row] = {
     val catalog = sparkSession.sessionState.catalog
     if (!catalog.tableExists(tableName)) {
-      logError(s"table '$tableName' in TRUNCATE TABLE does not exist.")
+      throw new AnalysisException(s"Table '$tableName' in TRUNCATE TABLE does not exist.")
```

Comment:
For drop table with IF EXISTS keyword, it is unfortunate that Hive does not throw an exception when the table does not exist and changing it breaks lots of tests in HiveCompatibilitySuite.